### PR TITLE
CIP-0105 | Add missing test vectors

### PR DESCRIPTION
**Pull request by paweljakubas**
Work remaining:

1. To be added : `x_sk` 
2. To verify if the keys/hashes are correct

### CODE REVIEW

**Comment by Ryun1** [comment]
Hey @paweljakubas 

Thanks for this,
I will go through each test vector and verify and also add missing keys.

File: CIP-0105/README.md [hunk 1]
```diff
@@ -237,6 +237,134 @@ This is because we have modeled DRep IDs and CC credentials on stake pool operat
...
+#### Test vector 1
+`mnemonic: test walk nut penalty hip pave soap entry language right filter choice`
+`accountIx: 0x80000000`
+```
+drep_xsk: drep_xsk14rjh4rs2dzm6k5xxe5f73cypzuv02pknfl9xwnsjws8a7ulp530xztarpdlyh05csw2cmnekths7dstq0se3wtza84m4fueffezsjfgassgs9xtfzgehrn0fn7c82uc0rkj06s0w0t80hflzz8cwyry3eg9066uj
+drep_sk:
+drep_xvk: drep_xvk17axh4sc9zwkpsft3tlgpjemfwc0u5mnld80r85zw7zdqcst6w543mpq3q2vkjy3nw8x7n8asw4es78dyl4q7u7kwlwn7yy0sugxfrjs6z25qe
+drep_vk: drep_vk17axh4sc9zwkpsft3tlgpjemfwc0u5mnld80r85zw7zdqcst6w54sdv4a4e
+drep: drep15k6929drl7xt0spvudgcxndryn4kmlzpk4meed0xhqe25nle07s
+drep_script_1: drep_script16pjhzfkm7rqntfezfkgu5p50t0mkntmdruwlp089zu8v29l95rg
```
Comment (Ryun1) [inline CIP-0105/README.md:260]:
```suggestion
```

the `drep_script` prefix is meant to be used for script-based DReps, including here I think is a mistake

Comment (paweljakubas) [reply CIP-0105/README.md:260]: this is the key:
```
drep_script16pjhzfkm7rqntfezfkgu5p50t0mkntmdruwlp089zu8v29l95rg
```
`drep_script_1` means we used the following script template:
```
script_1: all [keyhash, active_from 5001]
```
meaning as `drep` is acting as verification key hash we construct the following script:
```
all [drep15k6929drl7xt0spvudgcxndryn4kmlzpk4meed0xhqe25nle07s, active_from 5001]
```
serialize it and hash it, and then outputting with bech32 HRP "drep_script".

File: CIP-0105/README.md [hunk 2]
```diff
@@ -237,6 +237,134 @@ This is because we have modeled DRep IDs and CC credentials on stake pool operat
...
+cc_hot_sk:
+cc_hot_xvk: cc_hot_xvk10qawpxlz7eytt9yr4xlwtjkw345v0ehzsxdlkks6qralyp975phrzcymn4j2wypl4x43hnddlh4z6gmxkwlqy6xl0a5wmjdnd7xnqrsvak8ry
+cc_hot_vk: cc_hot_vk10qawpxlz7eytt9yr4xlwtjkw345v0ehzsxdlkks6qralyp975phqx538xn
+cc_hot: cc_hot1682whkcedz0ftcyhjxdasufyg85fks0vxm0y006qx38c2jz0ae0
+cc_hot_script_1: cc_hot_script1hheftszv4jw83f5megrvhrevl7lwwmtnjav7srkqngr92gna52t
+cc_hot_script_2: cc_hot_script1dg9jdwlsxzakctywv2cw7a7ggj2dwu0gz5tueu2rf40zvkj8dwc
+```
+
+#### Test vector 3
+`mnemonic: excess behave track soul table wearocean cash stay nature item turtlepalm soccer lunch horror start stumblemonth panic right must lock dress`
```
[the remark this answers is not in the document]
Comment (rphair) [reply CIP-0105/README.md]: https://github.com/cardano-foundation/CIPs/pull/840/commits/303b381c07a19f08e5e223c56430def9d473af26

[the remark this answers is not in the document]
Comment (paweljakubas) [reply CIP-0105/README.md]: indeed, error when removing "" during my processing -> corrected

File: CIP-0105/README.md [hunk 3]
```diff
@@ -237,6 +237,134 @@ This is because we have modeled DRep IDs and CC credentials on stake pool operat
...
+cc_hot_sk:
+cc_hot_xvk: cc_hot_xvk1tazd6lvnf2c9j8m58h6xy56uuyhkee526jgxj2ylaextl0xamd4nfjf0eunydl5kzvhk90aj5jhe92q3h6aph3laqpnpx0j7rhwtu9qe7dhsc
+cc_hot_vk: cc_hot_vk1tazd6lvnf2c9j8m58h6xy56uuyhkee526jgxj2ylaextl0xamd4swmuygc
+cc_hot: cc_hot1c2n5ax72vfqdj3ljn04hmmvkqjt5q9k694yw8f7rv3xvgxas90x
+cc_hot_script_1: cc_hot_script1tmwlec0twwvl29h6pgvew5mf4recsxtktev9g07xm37fv46mta9
+cc_hot_script_2: cc_hot_script1c77thg5lrahy0he4q6glsk8vgsp45gt75k3pq09d02u8g4s30yx
+```
+
+#### Test vector 4
+`mnemonic: excess behave track soul table wearocean cash stay nature item turtlepalm soccer lunch horror start stumblemonth panic right must lock dress`
```
[the remark this answers is not in the document]
Comment (paweljakubas) [reply CIP-0105/README.md]: indeed, error when removing "" during my processing -> corrected

**Review by rphair — COMMENTED** [review]
@paweljakubas looks fine from my own technical level and will ✅ following @Ryun1's endorsing it.

File: CIP-0105/README.md (hunk 1 above)
Comment (Ryun1) [reply CIP-0105/README.md:260]: ah okay, I understand now - thank you for the help!

File: CIP-0105/README.md [hunk 4]
```diff
@@ -237,6 +237,134 @@ This is because we have modeled DRep IDs and CC credentials on stake pool operat
...
+`script_1: all [keyhash, active_from 5001]`
+`script_2: any [keyhash, all [active_from 5001, active_until 6001]]`
+
+where
+
+`keyhash: {drep | cc_cold | cc_hot}`
+
+#### Test vector 1
+`mnemonic: test walk nut penalty hip pave soap entry language right filter choice`
+`accountIx: 0x80000000`
```
Comment (Ryun1) [inline CIP-0105/README.md:253]:
```suggestion
`accountIx: 0x00000000`
```
I believe this is a mistake.
The following key provided were produced using index of `0x00000000`.

Comment (Ryun1) [reply CIP-0105/README.md:253]: whoops, I was hardening the account index twice, disregard this

**Review by Ryun1 — APPROVED** [review]
This is a great addition to CIP-105
thank you for kicking this off @paweljakubas ✅

**Review by Crypto2099 — APPROVED** [review]
Given the feedback and review from @Ryun1 I feel comfortable marking this one as approved and getting changes merged.

**Review by rphair — APPROVED** [review]
Good all around; let's check it in...

**Pull request merged**